File: backend/app/models/supplier_model.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class SupplierCreate(BaseModel):
    nombre: str = Field(..., min_length=2)
    contacto: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    direccion: Optional[str] = None
    notas: Optional[str] = None
    activo: bool = True


class SupplierUpdate(BaseModel):
    nombre: Optional[str] = None
    contacto: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    direccion: Optional[str] = None
    notas: Optional[str] = None
    activo: Optional[bool] = None


class SupplierAdmin(BaseModel):
    supplier_id: int
    nombre: str
    contacto: Optional[str] = None
    telefono: Optional[str] = None
    email: Optional[str] = None
    direccion: Optional[str] = None
    notas: Optional[str] = None
    activo: bool
    created_at: datetime
    updated_at: datetime


class SupplierSummary(BaseModel):
    supplier_id: int
    nombre: str
    total_compras: float
    numero_compras: int

File: cash-control-app/lib/modules/business/models/supplier_model.dart
class SupplierModel {
  final int supplierId;
  final String nombre;
  final String? contacto;
  final String? telefono;
  final String? email;
  final String? direccion;
  final String? notas;
  final bool activo;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const SupplierModel({
    required this.supplierId,
    required this.nombre,
    this.contacto,
    this.telefono,
    this.email,
    this.direccion,
    this.notas,
    required this.activo,
    this.createdAt,
    this.updatedAt,
  });

  factory SupplierModel.fromJson(Map<String, dynamic> json) {
    return SupplierModel(
      supplierId: (json['supplier_id'] as num?)?.toInt() ?? 0,
      nombre: json['nombre']?.toString() ?? '',
      contacto: json['contacto']?.toString(),
      telefono: json['telefono']?.toString(),
      email: json['email']?.toString(),
      direccion: json['direccion']?.toString(),
      notas: json['notas']?.toString(),
      activo: json['activo'] as bool? ?? true,
      createdAt: _parseDate(json['created_at']),
      updatedAt: _parseDate(json['updated_at']),
    );
  }

  static DateTime? _parseDate(dynamic value) {
    if (value == null) return null;
    return DateTime.tryParse(value.toString());
  }
}

class SupplierSummaryModel {
  final int supplierId;
  final String nombre;
  final double totalCompras;
  final int numeroCompras;

  const SupplierSummaryModel({
    required this.supplierId,
    required this.nombre,
    required this.totalCompras,
    required this.numeroCompras,
  });

  factory SupplierSummaryModel.fromJson(Map<String, dynamic> json) {
    return SupplierSummaryModel(
      supplierId: (json['supplier_id'] as num?)?.toInt() ?? 0,
      nombre: json['nombre']?.toString() ?? '',
      totalCompras: (json['total_compras'] as num?)?.toDouble() ?? 0,
      numeroCompras: (json['numero_compras'] as num?)?.toInt() ?? 0,
    );
  }
}
